aurora/aurora-war/src/main/java/org/silverpeas/looks/aurora/AuroraSpaceHomePage.java

```diff
@@ -0,0 +1,372 @@
+    if (getData() != null) {
+      for (int i=1; i<=10; i++) {
+        Shortcut shortcut = getShortcut("shortcut", String.valueOf(i));
+        if (shortcut != null) {
+          shortcuts.add(shortcut);
+        }
+      }
+    }
+    return shortcuts;
+  }
+
+  private Shortcut getShortcut(String name, String nb) {
+    Shortcut shortcut = null;
+    String url = getFieldValue(name+"URL"+nb);
+    if (StringUtil.isDefined(url)) {
+      String icon = getFieldImageURL(name+"Icon"+nb);
+      String target = getFieldValue(name+"Target"+nb);
+      String alt = getFieldValue(name+"Label"+nb);
+      shortcut = new Shortcut(icon, target, url, alt);
+    }
+    return shortcut;
+  }
+
+  private boolean isEnabled(String lookParamName, String templateParamName) {
+    boolean enabled = (data == null && look.getSettings(lookParamName, true));
+    if (data != null) {
+      enabled = getFieldBooleanValue(templateParamName);
+    }
+    return enabled;
+  }
+
+  private void initBackOffice() {
+    ComponentInstLight app = look.getConfigurationApp(getSpace().getId());
+    if (app != null) {
+      setBackOfficeApp(app);
+      setData(getConfigurationData(app.getId()));
+    }
+  }
+
+  private ComponentInstLight getBackOfficeApp() {
+    return backOfficeApp;
+  }
+
+  private void setBackOfficeApp(final ComponentInstLight backOfficeApp) {
+    this.backOfficeApp = backOfficeApp;
+  }
+
+  private DataRecord getData() {
+    return data;
+  }
+
+  private void setData(final DataRecord data) {
+    this.data = data;
+  }
+
+  private DataRecord getConfigurationData(String appId) {
+    return getConfigurationData(appId, OrganizationController.get());
+  }
+
+  private DataRecord getConfigurationData(String appId, OrganizationController oc) {
+    String xmlFormName = oc.getComponentParameterValue(appId, "xmlTemplate");
+    if (!xmlFormName.isEmpty()) {
+      String xmlFormShortName =
+          xmlFormName.substring(xmlFormName.indexOf('/') + 1, xmlFormName.indexOf('.'));
+      try {
+        PublicationTemplate pubTemplate = PublicationTemplateManager.getInstance()
+            .getPublicationTemplate(appId + ":" + xmlFormShortName);
+        RecordSet recordSet = pubTemplate.getRecordSet();
+        DataRecord record = recordSet.getRecord("0", userLanguage);
+        if (record == null) {
+          record = recordSet.getEmptyRecord();
+          record.setId("0");
+          record.setLanguage(userLanguage);
+        }
+        return record;
+      } catch (Exception e) {
+        SilverLogger.getLogger(this).error(e);
+      }
+    }
+    return null;
+  }
+
+  private boolean getFieldBooleanValue(String name) {
+    return StringUtil.getBooleanValue(getFieldValue(name));
+  }
+
+  private int getFieldIntValue(String name) {
+    String value = getFieldValue(name);
+    if (StringUtil.isInteger(value)) {
+      return Integer.parseInt(value);
+    }
+    return 0;
+  }
+
+  private String getFieldImageURL(String name) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)